src/babashka/fs.cljc

```diff
@@ -670,14 +674,33 @@
   (.endsWith (as-path this) (as-path other)))
 
 (defn split-ext
-  "Splits a path into a vec of [name ext]. Works with strings, files, or paths."
+  "Splits a path into a vec of [path-without-ext ext]. Works with strings, files, or paths."
   [path]
   (let [name (file-name path)
         i (str/last-index-of name ".")
         ext (when (and i (pos? i)) (subs name (+ 1 i)))]
     (if ext
-      [(subs name 0 i) ext]
-      [name nil])))
+      (let [new-name (subs name 0 i)
+            new-path (-> (parent path) (path* new-name) str)]
+        [new-path ext])
+      [path nil])))
+
+(defn strip-ext
+  "Returns the path with the extension removed. If provided, a specific extension will be removed."
+  ([path]
+   (-> path split-ext first))
+  ([path extension]
+   (let [name (file-name path)
+         extension (if (str/starts-with? extension ".")
```

Comment:
Let's just make a convention that the function must be called with the extension without a dot, so we don't have to be clever here.